debian/Makefile

```diff
@@ -0,0 +1,152 @@
+# Dependencies you'll probably need to install to compile this: make, curl, git,
+# zip, unzip, patch, java8-jdk | openjdk-8-jdk, maven.
+
+SHELL=/bin/bash
+
+ifndef VERSION
+# Note that this is sensitive to this package's version being the first
+# <version> tag in the pom.xml
+VERSION=$(shell grep version pom.xml | head -n 1 | awk -F'>|<' '{ print $$3 }')
+endif
+
+export PACKAGE_TITLE=ksql
+export FULL_PACKAGE_TITLE=confluent-ksql
+export PACKAGE_NAME=$(FULL_PACKAGE_TITLE)-$(VERSION)
+
+# Defaults that are likely to vary by platform. These are cleanly separated so
+# it should be easy to maintain altered values on platform-specific branches
+# when the values aren't overridden by the script invoking the Makefile
+
+# Whether we should apply patches. This only makes sense for alternate packaging
+# systems that know how to apply patches themselves, e.g. Debian.
+APPLY_PATCHES?=yes
+
+# DESTDIR may be overriden by e.g. debian packaging
+ifeq ($(DESTDIR),)
+DESTDIR=$(CURDIR)/BUILD/
+endif
+
+ifeq ($(PACKAGE_TYPE),archive)
+PREFIX=$(PACKAGE_NAME)
+SYSCONFDIR=$(PREFIX)/etc/$(PACKAGE_TITLE)
+SYSTEMDDIR=$(PREFIX)/lib/systemd/system
+else
+PREFIX=/usr
+SYSCONFDIR=/etc/$(PACKAGE_TITLE)
+SYSTEMDDIR=/lib/systemd/system
+endif
+
+# Whether we should run tests during the build.
+SKIP_TESTS?=yes
+
+all: install
+
+
+archive: install
+	rm -f $(CURDIR)/$(PACKAGE_NAME).tar.gz && cd $(DESTDIR) && tar -czf $(CURDIR)/$(PACKAGE_NAME).tar.gz $(PREFIX)
+	rm -f $(CURDIR)/$(PACKAGE_NAME).zip && cd $(DESTDIR) && zip -r $(CURDIR)/$(PACKAGE_NAME).zip $(PREFIX)
+
+apply-patches: $(wildcard patches/*)
+ifeq ($(APPLY_PATCHES),yes)
+	git reset --hard HEAD
+	cat patches/series | xargs -iPATCH bash -c 'patch -p1 < patches/PATCH'
+endif
+
+build: apply-patches
+ifeq ($(SKIP_TESTS),yes)
+	mvn -DskipTests=true install
+else
+	mvn install
+endif
+
+BINPATH=$(PREFIX)/bin
+LIBPATH=$(PREFIX)/share/java/$(PACKAGE_TITLE)
+DOCPATH=$(PREFIX)/share/doc/$(PACKAGE_TITLE)
+
+INSTALL=install -D -m 644
+INSTALL_X=install -D -m 755
+
+install: build
+	# Safety precatuion to avoid removing of root dir when DESTDIR or PREFIX is not set, for wathever reason
+	if [[ $(DESTDIR)$(PREFIX) != /tmp/confluent/* ]]; then echo "DESTDIR=$(DESTDIR) or PREFIX=$(PREFIX) is weird" ; exit 1 ; fi
+	rm -rf $(DESTDIR)$(PREFIX)
+	mkdir -p $(DESTDIR)$(PREFIX)
+	mkdir -p $(DESTDIR)$(BINPATH)
+	mkdir -p $(DESTDIR)$(LIBPATH)
+	mkdir -p $(DESTDIR)$(SYSCONFDIR)
+	for svc in debian/*.service ; do \
+	    $(INSTALL) -o root -g root -D $$svc $(DESTDIR)$(SYSTEMDDIR)/$$(basename $$svc) ; \
+	done
+	pushd "ksql/target/ksql-$(VERSION)-package" ; \
+	find bin/ -type f | grep -v README[.]rpm | xargs -I XXX $(INSTALL_X) -o root -g root XXX $(DESTDIR)$(PREFIX)/XXX ;\
+	find share/ -type f | grep -v README[.]rpm | xargs -I XXX $(INSTALL) -o root -g root XXX $(DESTDIR)$(PREFIX)/XXX ; \
+	pushd etc/ksql/ ; \
+	find . -type f | grep -v README[.]rpm | xargs -I XXX $(INSTALL) -o root -g root XXX $(DESTDIR)$(SYSCONFDIR)/XXX
```

Comment:
Verify final package contents with:

for archive:
 tar tvzf output/..ksql..tar.gz

rpm:
  rpm -qpl output/..ksql..rpm

deb:
  dpkg -c output/..ksql..all.deb